src/error.rs

```diff
@@ -13,101 +48,101 @@ use core::fmt;
 macro_rules! write_err {
     ($writer:expr, $string:literal $(, $args:expr)*; $source:expr) => {
         {
-            #[cfg(feature = "std")]
-            {
-                let _ = &$source;   // Prevents clippy warnings.
-                write!($writer, $string $(, $args)*)
-            }
-            #[cfg(not(feature = "std"))]
-            {
-                write!($writer, concat!($string, ": {}") $(, $args)*, $source)
+            if_std_error! {
+                {
+                    {
+                        let _ = &$source;   // Prevents clippy warnings.
+                        write!($writer, $string $(, $args)*)
+                    }
+                } else {
+                    {
+                        write!($writer, concat!($string, ": {}") $(, $args)*, $source)
+                    }
+                }
             }
         }
     }
 }
+pub(crate) use write_err;
 
-/// Hex decoding error.
+/// Error returned when hex decoding a hex string with variable length.
+///
+/// This represents the first error encountered during decoding, however we may add other remaining
+/// ones in the future.
+///
+/// This error differs from `DecodeFixedSizedBytesError` in that the number of bytes is only known at run time
+/// - e.g. when decoding `Vec<u8>`.
```

Comment:
This renders as a list item because of the `-` at the start of the line.

Reply:
We could use a link for `DecodeFixedSizedBytesErro` after 'differs from' since reader is likely to want to see the `DecodeFixedSizedBytesError` when reading.

Reply:
This fixes both things

```rust
/// This error differs from [`DecodeDynSizedBytesError`] in that the number of bytes is known at
/// compile time - e.g. when decoding to an array of bytes.
```